conans/client/build/cmake.py

```diff
@@ -307,9 +307,17 @@ def build(self):
                     # current package
                     path_content = tools.load(path)
                     for dep in self._conanfile.deps_cpp_info.deps:
-                        from_str = self._conanfile.deps_cpp_info[dep].rootpath
+                        from_str = self._format_path_to_cmake_config(self._conanfile.deps_cpp_info[dep].rootpath)
                         # try to replace only if from str is found
                         if path_content.find(from_str) != -1:
                             dep_str = "${CONAN_%s_ROOT}" % dep.upper()
                             self._conanfile.output.info("Patching paths for %s: %s to %s" % (dep, from_str, dep_str))
                             tools.replace_in_file(path, from_str, dep_str, strict=False)
+
+    def _format_path_to_cmake_config(self, pathstr):
+        if self._conanfile.settings.os == "Windows":
+            drive,path = os.path.splitdrive(pathstr)
+            return drive.upper() + path.replace(os.path.sep,"/")
+        return pathstr.replace(os.path.sep,"/")
```

Comment:
So, AFAIK `os.path.sep == "/"` if not windows, so the path will remain exactly the same?

Reply:
Maybe we can just return the path when `self._conanfile.settings.os != "Windows"` otherwise the current code.

Reply:
I was under the impression that conan was aiming to support RISC OS  ("." separator) and mac OS 9 (":" separator) :-P
Seriously though, yes I believe this is the case and you're correct. 
In fact, having now used this feature in quite a few libraries, what I have discovered is that 99% of the time on linux/OSX the cmake files are relocatable anyway. It's really only on windoiws, where the export handler generates by default a line like

` set(_IMPORT_PREFIX "C:\some\absolute\path") `